Makefile

```diff
@@ -26,4 +34,8 @@ clean:
 	${MAKE} -C ${BUILD_DIR} clean
 
 clobber:
+ifeq ($(wildcard ${BUILD_DIR}/CMakeCache.txt),)
+	$(error CMakeCache.txt does not exist in the specified BUILD_DIR. You might be pointing \
+	to the wrong directory to be cleaned or you have not even built the sources yet)
+endif
```

Comment:
This will potentially break workflows that unconditionally do "make clobber" before a build to ensure a clean build from scratch. This could perhaps be a warning or just a plain status message instead of an error.

Reply:
This is tricky, because I give the user the opportunity to specify BUILD_DIR. If a user would somehow specify BUILD_DIR as /, e. g. "make BUILD_DIR=/ clobber", he would wipe his system. And with the beforehand solution, "make -f some/dir/in/system/OpenCL-ICD-Loader" would also have this devastating result.

Reply:
I try to come up with a solution

Reply:
I changed the assertion to a warning and added a prompt to the user. Changes are ready for review.